src/utils/Process.ts

```diff
@@ -0,0 +1,102 @@
+import {executeCommandWithShellConfig} from "../PTY";
+import {/*memoize, */intersection} from "lodash";
+
+// http://linuxcommand.org/man_pages/ps1.html
+//   ps -eo euser,ruser,euid,ruid,comm
+//   ps -eo rgid,egid,rgroup,egroup,comm
+
+export interface User {
+    ruserid: string;
+    ruser: string;
+    euserid: string;
+    euser: string;
+}
+
+export interface Group {
+    rgroupid: string;
+    rgroup: string;
+    egroupid: string;
+    egroup: string;
+}
+
+export interface Terminal {
+    name: string;
+}
+
+export interface Process {
+    pid: string;
+    time: string;
+    ruser: string;
+    cmd: string;
+}
+
+export interface Session {
+    sid: string;
+    rgroup: string;
+}
+
+const ignoreUsers: string[] = ["nobody"];
+
+const ignoreGroups: string[] = ["nobody"];
+
+export const users =
+    async(): Promise<User[]> => {
+        const pInfo: string[] = await executeCommandWithShellConfig("ps -eo ruid,ruser,euid,euser");
+        return pInfo.splice(1)
+                    .map(p => {
+                        const v = p.trim().replace(/ +(?= )/g, "").split(" ");
```

Comment:
`p.trim().replace(/ +(?= )/g, "").split(" ")` is repeated in a bunch of places, can you extract that into a function?